vic/drivers/shared_image/src/vic_restore.c

```diff
@@ -0,0 +1,1009 @@
+    extern option_struct    options;
+    extern soil_con_struct *soil_con;
+
+    int                     status;
+    size_t                  dimlen;
+    size_t                  i;
+    size_t                  d1count[1];
+    size_t                  d1start[1];
+    size_t                  d2count[2];
+    size_t                  d2start[2];
+    int                     lon_var_id;
+    int                     lat_var_id;
+    double                 *dvar;
+    nc_file_struct          nc;
+
+    // open the netcdf file
+    status = nc_open(filenames.init_state, NC_SHARE, &(nc.nc_id));
+    if (status != NC_NOERR) {
+        log_err("Error opening %s", filenames.init_state);
+    }
+
+    // read and validate dimension lengths
+    dimlen = get_nc_dimension(filenames.init_state, "ni");
+    if (dimlen != global_domain.n_nx) {
+        log_err("Number of grid columns in state file does not "
+                "match parameter file");
+    }
+    dimlen = get_nc_dimension(filenames.init_state, "nj");
+    if (dimlen != global_domain.n_ny) {
+        log_err("Number of grid rows in state file does not "
+                "match parameter file");
+    }
+    dimlen = get_nc_dimension(filenames.init_state, "veg_class");
+    if (dimlen != options.NVEGTYPES) {
+        log_err("Number of veg classes in state file does not "
+                "match parameter file");
+    }
+    dimlen = get_nc_dimension(filenames.init_state, "snow_band");
+    if (dimlen != options.SNOW_BAND) {
+        log_err("Number of snow bands in state file does not "
+                "match parameter file");
+    }
+    dimlen = get_nc_dimension(filenames.init_state, "nlayer");
+    if (dimlen != options.Nlayer) {
+        log_err("Number of soil layers in state file does not "
+                "match parameter file");
+    }
+    dimlen = get_nc_dimension(filenames.init_state, "frost_area");
+    if (dimlen != options.Nfrost) {
+        log_err("Number of frost areas in state file does not "
+                "match parameter file");
+    }
+    dimlen = get_nc_dimension(filenames.init_state, "soil_node");
+    if (dimlen != options.Nnode) {
+        log_err("Number of soil nodes in state file does not "
+                "match parameter file");
+    }
+    if (options.LAKES) {
+        dimlen = get_nc_dimension(filenames.init_state, "lake_node");
+        if (dimlen != options.NLAKENODES) {
+            log_err("Number of lake nodes in state file does not "
+                    "match parameter file");
+        }
+    }
+
+    // read dimension variables
+
+    // lat/lon
+    status = nc_inq_varid(nc.nc_id, "lon", &lon_var_id);
+    if (status != NC_NOERR) {
+        log_err("Unable to find variable \"lon\" in %s",
+                filenames.init_state);
+    }
+    status = nc_inq_varid(nc.nc_id, "lat", &lat_var_id);
+    if (status != NC_NOERR) {
+        log_err("Unable to find variable \"lat\" in %s",
+                filenames.init_state);
+    }
```

Comment:
You mean lon/lat_var_id?  Wouldn't these be potentially different from file to file, i.e. wouldn't global_domain->info's versions of them be coming from some other file like the forcings or parameters?  I'm assuming these ids reflect the order in which they're defined in the file at the time of file creation.  If someone creates an initial state file outside of VIC or tonic, are we going to penalize them for not having defined their dimensions in the exact same order as VIC?  Or am I missing something here?